Dockerfile

```diff
@@ -0,0 +1,31 @@
+# select build image
+FROM rust as builder
+
+# create a new empty shell project
+RUN USER=root cargo new --bin trin
+WORKDIR /trin
+
+RUN apt-get update && apt-get install clang -y
+RUN rustup component add rustfmt
```

Comment:
Just curious: why do we need rustfmt in this docker builder?

Reply:
Good catch, this was necessary in an earlier, fancier setup I was attempting when I was running `cargo build` in default mode (which required rustfmt to compile) - but it's no longer necessary so I've removed it